projects/igniteui-angular/src/lib/directives/input/input.directive.ts

```diff
@@ -349,16 +349,23 @@ export class IgxInputDirective implements AfterViewInit, OnDestroy {
     protected updateValidityState() {
         if (this.ngControl) {
             if (this.ngControl.control.validator || this.ngControl.control.asyncValidator) {
+                // Run the validation with empty object to check if required is enabled.
+                const error = this.ngControl.control.validator({} as AbstractControl);
+                this.inputGroup.isRequired = error && error.required;
```

Comment:
Just figured this is setting only to the group which will create the correct UI (asterisk) but won't reflect on the input itself in terms of accessibility. Perhaps something to look into for https://github.com/IgniteUI/igniteui-angular/issues/9996